### PR TITLE
Standard error raised for failing company name

### PR DESCRIPTION
**Pull request by Beckyrose200**
https://eaflood.atlassian.net/browse/RUBY-1877 
Here we have added some logic in the refresh company house service to throw an error if the company name is returning nil.

### CODE REVIEW

File: app/services/waste_exemptions_engine/refresh_companies_house_name_service.rb [hunk 1]
```diff
@@ -7,6 +7,8 @@ class RefreshCompaniesHouseNameService < WasteExemptionsEngine::BaseService
     def run(company_reference)
       registration = Registration.find_by(reference: company_reference)
       company_name = DefraRubyCompaniesHouse.new(registration.company_no).company_name
+      raise StandardError if company_name == nil?
```
Comment (PaulDoyle-DEFRA) [inline app/services/waste_exemptions_engine/refresh_companies_house_name_service.rb:10]: Use `.new` and give the exception a message, e.g: `raise StandardError.new "Company name not found"`, or just `raise "Company name not found"`

File: app/services/waste_exemptions_engine/refresh_companies_house_name_service.rb [hunk 2]
```diff
@@ -7,6 +7,8 @@ class RefreshCompaniesHouseNameService < WasteExemptionsEngine::BaseService
     def run(company_reference)
       registration = Registration.find_by(reference: company_reference)
       company_name = DefraRubyCompaniesHouse.new(registration.company_no).company_name
```
Comment (PaulDoyle-DEFRA) [inline app/services/waste_exemptions_engine/refresh_companies_house_name_service.rb:9]: What type of exception does RestClient raise if the URL is invalid or the request times out? Need to handle that as well.

**Review by PaulDoyle-DEFRA — CHANGES_REQUESTED** [review]
Need to handle other exception types including timeouts and URL errors, possibly in DefraRubyCompaniesHouse.

File: lib/defra_ruby_companies_house.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@ def initialize(company_no)
     @company_url = "#{Rails.configuration.companies_house_host}#{company_no.to_s.rjust(8, '0')}"
     @api_key = Rails.configuration.companies_house_api_key
 
-    load_company
+    StandardError unless load_company
```
Comment (PaulDoyle-DEFRA) [inline lib/defra_ruby_companies_house.rb:10]: `raise StandardError unless load_company`

File: spec/lib/defra_ruby_companies_house_spec.rb [hunk 4]
```diff
@@ -70,4 +70,14 @@
       end
     end
   end
+
+  context "when the service is down" do
+    before do
+      allow_any_instance_of(DefraRubyCompaniesHouse).to receive(:load_company).and_raise(StandardError)
```
Comment (PaulDoyle-DEFRA) [inline spec/lib/defra_ruby_companies_house_spec.rb:76]: I think this should be raising one of the RestClient exception types - the test is to confirm that it get mapped to StandardError

File: lib/defra_ruby_companies_house.rb (hunk 3 above)
Comment (Beckyrose200) [reply lib/defra_ruby_companies_house.rb:10]: image
Putting raise in makes a lot of the rspec tests fail

File: spec/lib/defra_ruby_companies_house_spec.rb (hunk 4 above)
Comment (PaulDoyle-DEFRA) [reply spec/lib/defra_ruby_companies_house_spec.rb:76]: Actually there should be one test each for (a) when the company number is not valid (ResourceNotFound) and (b) when the service is down (NotFound)

**Review by PaulDoyle-DEFRA — CHANGES_REQUESTED** [review]
Apologies for noticing earlier, but I think RefreshCompaniesHouseNameService should be in the back-office, not the engine, because it is back-office only functionality.

**Review by PaulDoyle-DEFRA — APPROVED** [review]

**Pull request merged**